client.go

```diff
@@ -204,15 +204,18 @@ func (client *Client) getWarrants() (WarrantSet, error) {
 		return nil, errors.New(fmt.Sprintf("received HTTP %d: %s", respStatus, string(msg)))
 	}
 
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, errors.Wrap(err, "error reading response from server")
-	}
+	warrants := make(WarrantSet)
+	decoder := json.NewDecoder(resp.Body)
+	for decoder.More() {
+		var chunk WarrantSet
+		err := decoder.Decode(&chunk)
+		if err != nil {
+			return nil, errors.Wrap(err, "error reading response from server")
```

Comment:
Just confirm that returning an error here closes the response body (might do it automatically via the decoder)